packages/core/src/lib/http-interceptor-utils.ts

```diff
@@ -1,42 +1,55 @@
 import kleur from 'kleur';
 import { BatchInterceptor } from '@mswjs/interceptors';
+import { FetchInterceptor } from '@mswjs/interceptors/fetch';
 // import { ClientRequestInterceptor } from '@mswjs/interceptors/ClientRequest';
 // import { XMLHttpRequestInterceptor } from '@mswjs/interceptors/XMLHttpRequest';
-import { FetchInterceptor } from '@mswjs/interceptors/fetch';
 
 import { checkUrlInAllowList } from './url-pattern-utils';
-import { SensitiveValueLookup } from '../config-engine/config-engine';
+import type { SensitiveValueLookup } from '../config-engine/config-engine';
 
 
 // some funky stuff happening to keep track of the interceptor globally
 // because during vite reloads the module may be reloaded multiple times
 
 function initInterceptor() {
-  if ((globalThis as any)._dmnoHttpInterceptor) {
+  const fetchAlreadyPatched = Object.getOwnPropertySymbols(globalThis.fetch)?.[0]?.toString() === 'Symbol(isPatchedModule)';
+  // console.log('init interceptor', globalThis.fetch, `patched = ${fetchAlreadyPatched}`, `global interceptor? ${!!(globalThis as any)._dmnoHttpInterceptor}`);
+  if (fetchAlreadyPatched) {
+    // console.log('already patched');
     return;
   }
+  if ((globalThis as any)._dmnoHttpInterceptor) {
+    // console.log('disposing existing interceptor');
+    (globalThis as any)._dmnoHttpInterceptor.dispose();
```

Comment:
this is what was broken in netlify, but I suspect it might be breaking something else (like nextjs + edge?). should be uncovered when we get the full test suite set up.